examples/volumes/flexvolume/README.md

```diff
@@ -1,54 +1,94 @@
 # Flexvolume
 
-Flexvolume enables users to mount vendor volumes into kubernetes. It expects vendor drivers are installed in the volume plugin path on every kubelet node.
-
-It allows for vendors to develop their own drivers to mount volumes on nodes.
+Flexvolume enables users to write their own drivers and add support for their volumes in Kubernetes. Vendor drivers should be installed in the volume plugin path on every Kubelet node and on master node(s) if "-enable-controller-attach-detach" Kubelet option is enabled. 
 
 *Note: Flexvolume is an alpha feature and is most likely to change in future*
 
 ## Prerequisites
 
-Install the vendor driver on all nodes in the kubelet plugin path. Path for installing the plugin: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/\<vendor~driver\>/\<driver\>
+Install the vendor driver on all nodes (also on master nodes if "-enable-controller-attach-detach" Kubelet option is enabled) in the plugin path. Path for installing the plugin: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/\<vendor~driver\>/\<driver\>
 
 For example to add a 'cifs' driver, by vendor 'foo' install the driver at: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/\<foo~cifs\>/cifs
 
 ## Plugin details
-
-Driver will be invoked with 'Init' to initialize the driver. It will be invoked with 'attach' to attach the volume and with 'detach' to detach the volume from the kubelet node. It also supports custom mounts using 'mount' and 'unmount' callouts to the driver.
+The plugin expects the following call-outs are implemented for the backend drivers. Some call-outs are optional. Call-outs are invoked from the Kubelet & Controller manager nodes.
 
 ### Driver invocation model:
 
-Init:
+#### Init:
+Initializes the driver.
 
 ```
 <driver executable> init
 ```
 
-Attach:
+#### Get volume name:
+Get a cluster wide unique volume name for the volume.
+
+```
+<driver executable> getvolumename <json options>
+```
+
+#### Attach:
+Attach the volume specified by the given spec on the given host. On success, returns the device path where the device is attached on the node. Nodename param is only valid if "-enable-controller-attach-detach" Kubelet option is enabled.
```

Comment:
>  Nodename param is only valid if "-enable-controller-attach-detach" Kubelet option is enabled.

When attach/detach controller is disabled, kubelet calls attach, and I believe still provides the node name, no?